### DISCUSSION TITLE
Babel 7.16 ends up with babel 7.18 dependencies

### DISCUSSION THREAD

**Discussion by cyqui** [babel/babel#14725] [General] [answered]
Hello,

We had a project that was using babel/register with mocha silently exiting.
After few hours of debug, i ended up fixing it by using npm "override" to fixate babel version at 7.16 and it did the trick for some reason...

I also noticed that the following package.json seem to end up with mixed versions.

```
{
  "name": "babel",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@babel/register": "7.16.0",
    "@babel/cli": "7.16.0",
    "@babel/core": "7.16.0",
    "@babel/node": "7.16.0",
    "@babel/preset-env": "7.16.0"
  }
}
```

output of npm list --depth=1

```
babel@1.0.0 /Users/cyqui/discuss/babel
├─┬ @babel/cli@7.16.0
│ ├── @babel/core@7.16.0 deduped
....
├─┬ @babel/core@7.16.0
│ ├── @babel/code-frame@7.18.6 <========================== 7.18.6 ?
│ ├── @babel/generator@7.18.7<========================== 7.18.6
│ ├── @babel/helper-compilation-targets@7.18.6<========================== 7.18.6
│ ├── @babel/helper-module-transforms@7.18.6<========================== 7.18.6
│ ├── @babel/helpers@7.18.6<========================== 7.18.6
│ ├── @babel/parser@7.18.6<========================== 7.18.6
│ ├── @babel/template@7.18.6
│ ├── @babel/traverse@7.18.6
│ ├── @babel/types@7.18.7
```
Anyone has an opinion ? is that normal ?

**Comment by JLHwung** [comment, marked as answer]
It is normal, Babel follows semantic versioning, which means`@babel/core` 7.16.0 should be compatible with `@babel/code-frame` 7.18.6 and so on. If you spot any compatibility issue between `@babel/core` 7.16.0, please prepare a git repo so we can clone and reproduce the errors, and file an issue to https://github.com/babel/babel/issues.

**Reply by cyqui** [reply]
ok ! will try to make a repo. i couldn't understand what was exactly the problem but having everything on 7.16 fixes the issue.
